code/src/nuvla/connector/docker_cli_connector.py

```diff
@@ -123,17 +138,31 @@ def _extract_service_info(stack_name, service):
         service_info['replicas.running'] = replicas_running
         ports = filter(None, service_json['Ports'].split(','))
         for port in ports:
-            external_port_info, internal_port_info = port.split('->')
-            external_port = external_port_info.split(':')[1]
-            internal_port, protocol = internal_port_info.split('/')
+            protocol, internal_port, external_port = DockerCliConnector.ports_info(port)
             service_info['{}.{}'.format(protocol, internal_port)] = external_port
+
+        # Check if service has host ports published
```

Comment:
I don't want to parse docker-compose files because of the different syntax of it to put ports mapping

Reply:
is there another syntax for host mode?

Reply:
After check, you have to use the long syntax to define host mode. But I'm not sure it will be faster to parse a yaml file or to make a request to list tasks.